dockers/docker-gnmi-watchdog/watchdog/src/main.rs

```diff
@@ -0,0 +1,86 @@
+use std::io::{Read, Write};
+use std::net::TcpListener;
+use std::process::Command;
+use chrono::{TimeZone};
+
+static NSENTER_CMD: &str = "nsenter --target 1 --pid --mount --uts --ipc --net";
+
+// Helper to run commands
+fn run_command(cmd: &str) -> Result<String, String> {
+    let output = Command::new("sh")
+        .arg("-c")
+        .arg(cmd)
+        .output()
+        .map_err(|e| format!("Failed to spawn command '{}': {}", cmd, e))?;
+
+    if !output.status.success() {
+        return Err(format!(
+            "Command '{}' failed with status {}: {}",
+            cmd,
+            output.status.code().map_or("unknown".to_string(), |c| c.to_string()),
+            String::from_utf8_lossy(&output.stderr).to_string() // Clone the error message
+        ));
+    }
+
+    Ok(String::from_utf8_lossy(&output.stdout).to_string()) // Clone the output
+}
+
+// Check gnmi program status
+fn check_gnmi_status() -> String {
+
+    "OK".to_string()
+}
+
+fn main() {
+    let watchdog_port = 51000;
```

Comment:
>watchdog_port [](http://example.com/codeflow?start=8&length=13)

If the port occupied, will the it break container upgrade? let's at least manual test it.